iam_builder/templates.py

```diff
@@ -6,207 +6,219 @@
-                "glue:DeleteTable",
-                "glue:UpdateDatabase",
-                "glue:UpdatePartition",
-                "glue:UpdateTable",
-                "glue:CreateUserDefinedFunction",
-                "glue:DeleteUserDefinedFunction",
-                "glue:UpdateUserDefinedFunction"
-            ],
-            "Resource": [
-                "*"
-            ]
-        }
-    ],
-    "glue_job": [
-        {
-            "Sid": "GlueJobActions",
-            "Effect": "Allow",
-            "Action": [
-                "glue:BatchStopJobRun",
-                "glue:CreateJob",
-                "glue:DeleteJob",
-                "glue:GetJob",
-                "glue:GetJobs",
-                "glue:GetJobRun",
-                "glue:GetJobRuns",
-                "glue:StartJobRun",
-                "glue:UpdateJob",
-                "glue:ListJobs",
-                "glue:BatchGetJobs",
-                "glue:GetJobBookmark"
-            ],
-            "Resource": [
-                "*"
-            ]
-        },
-        {
-            "Sid": "CanGetLogs",
-            "Effect": "Allow",
-            "Action": [
-                "logs:GetLogEvents",
-                "logs:CreateLogGroup",
-                "logs:CreateLogStream",
-                "logs:PutLogEvents",
-                "logs:DescribeLogStreams"
-            ],
-            "Resource": [
-                "arn:aws:logs:*:*:/aws-glue/*"
-            ]
-        },
-        {
-            "Sid": "CanGetCloudWatchLogs",
-            "Effect": "Allow",
-            "Action": [
-                "cloudwatch:PutMetricData",
-                "cloudwatch:GetMetricData",
-                "cloudwatch:ListDashboards"
-            ],
-            "Resource": [
-                "*"    
-            ]
-        },
-        {
-            "Sid": "CanReadGlueStuff",
-            "Effect": "Allow",
-            "Action": [
-                "s3:GetObject",
-                "s3:PutObject"
-            ],
-            "Resource": [
-                "arn:aws:s3:::aws-glue-*/*",
-                "arn:aws:s3:::*/*aws-glue-*/*",
-                "arn:aws:s3:::aws-glue-*"
-            ]
-        }
-    ],
-    "decrypt_statement": [
-        {
-            "Sid": "allowDecrypt",
-            "Effect": "Allow",
-            "Action": [
-                "kms:Decrypt"
-            ],
-            "Resource": [
-                "arn:aws:kms:::key/*"
-            ]
-        }
+def iam_lookup(athena_dump_bucket):
+    """Creates a full lookup policy based on the selected dump bucket
+
+    Parameters
+    ----------
+    athena_dump_bucket (str/list): bucket name or names for Athena query dump
+    """
+    # prepare sections of iam lookup that depend on dump bucket or buckets
+    allow_list_bucket_resources = ["arn:aws:s3:::moj-analytics-lookup-tables"]
+    allow_list_bucket_resources.extend([
+        "arn:aws:s3:::" + bucket for bucket in athena_dump_bucket
+        ])
```

Comment:
Ah yeah, it will on its own, but the bit of iam_builder that calls this function will turn "text" into ["text"], avoiding this. But this isn't a very clear way of doing it so I'll tidy it up - quite possibly moving it into this function.

Reply:
Made that function list only (and hopefully the type hints will clarify further)